src/renderer/components/item-object-list/item-list-layout.tsx

```diff
@@ -397,20 +390,26 @@ export class ItemListLayout extends React.Component<ItemListLayoutProps> {
   }
 
   renderHeader() {
-    const { showHeader, customizeHeader, renderHeaderTitle, headerClassName, isClusterScoped } = this.props;
+    const { showHeader, customizeHeader, renderHeaderTitle, headerClassName } = this.props;
+
+    if (!showHeader) {
+      return null;
+    }
 
-    if (!showHeader) return null;
+    const showNamespaceSelectFilter = this.props.store instanceof KubeObjectStore && this.props.store.api.isNamespaced;
     const title = typeof renderHeaderTitle === "function" ? renderHeaderTitle(this) : renderHeaderTitle;
     const placeholders: IHeaderPlaceholders = {
       title: <h5 className="title">{title}</h5>,
       info: this.renderInfo(),
-      filters: <>
-        {!isClusterScoped && <NamespaceSelectFilter/>}
-        <PageFiltersSelect allowEmpty disableFilters={{
-          [FilterType.NAMESPACE]: true, // namespace-select used instead
-        }}/>
-      </>,
-      search: <SearchInputUrl/>,
+      filters: (
+        <>
+          {showNamespaceSelectFilter && <NamespaceSelectFilter />}
```

Comment:
We might want to get rid of the default (namespace) filter in future. ItemListLayout should be "item agnostic" and Kube object list can set this easily.

Reply:
That is what the follow up PRs are doing.